src/NServiceBus.Persistence.DynamoDB.PersistenceTests/When_saving_saga_with_empty_list.cs

```diff
@@ -0,0 +1,173 @@
+﻿namespace NServiceBus.PersistenceTesting;
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Sagas;
+
+public class When_saving_saga_with_empty_list : SagaPersisterTests
+{
+    [Test]
+    public async Task ShouldSave()
+    {
+        var sagaData = new EmptyCollectionsSagaData { SomeId = Guid.NewGuid().ToString() };
+        await SaveSaga(sagaData);
+
+        var read = await GetById<EmptyCollectionsSagaData>(sagaData.Id);
+        Assert.That(read, Is.Not.Null);
+        Assert.That(read.SomeId, Is.EqualTo(sagaData.SomeId));
+
+        Assert.Multiple(() =>
+        {
+            Assert.That(read.StringList, Is.Empty);
+            Assert.That(read.StringArray, Is.Empty);
+            Assert.That(read.RecordList, Is.Empty);
+            Assert.That(read.RecordArray, Is.Empty);
+            Assert.That(read.SimpleDict, Is.Empty);
+        });
+    }
+
+    [Test]
+    public async Task ShouldUpdate()
+    {
+        var sagaData = new EmptyCollectionsSagaData
+        {
+            SomeId = Guid.NewGuid().ToString(),
+            StringList = ["a", "b"],
+            StringArray = ["c", "d"],
+            RecordList = [new("e", 1.2), new("f", 3.4)],
+            RecordArray = [new("g", 5.6), new("h", 7.8)],
+            SimpleDict = new Dictionary<string, int>
+            {
+                ["i"] = 9,
+                ["j"] = 10,
+            },
+            Ints = [11, 12],
+            Doubles = [13.4, 15.6],
```

Comment:
By looking at this test I think we should be consistent and fill in values to all properties that we then also reassign to empty. Otherwise, it is confusing why we set some and don't set others.